### PR TITLE
Migrate to matterjs

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

File: src/matter/cluster/AccessControlCluster.ts [hunk 1]
```diff
@@ -80,40 +80,50 @@ export const enum ChangeTypeEnum {
  *
  * @see {@link MatterCoreSpecificationV1_0} § 9.10.5.3
  */
```
Comment (mfucci) [inline src/matter/cluster/AccessControlCluster.ts:82]: This is one way to use it: deconstruct tlv into the base classes and reference them directly.

File: src/matter/cluster/AccessControlCluster.ts [hunk 2]
```diff
@@ -80,40 +80,50 @@ export const enum ChangeTypeEnum {
...
-    endpoint: Field(1, EndpointNumberT), /* nullable: true */
+    endpoint: Field(1, Nullable(EndpointNumberT)),
+
     /** Device type to grant access on. */
-    deviceType: Field(1, DeviceTypeIdT), /* nullable: true */
+    deviceType: Field(1, Nullable(DeviceTypeIdT)),
 });
 
 /** @see {@link MatterCoreSpecificationV1_0} § 9.10.5.3 */
-const AccessControlEntryT = ObjectT({
```
Comment (mfucci) [inline src/matter/cluster/AccessControlCluster.ts:93]: This is a 2nd way: keep the name space

File: src/matter/cluster/AccessControlCluster.ts [hunk 3]
```diff
@@ -128,47 +138,40 @@ const AccessChangeEvent = <T>(entryTemplate: Template<T>) => ({
...
          * The cluster SHALL send AccessControlEntryChanged events whenever its ACL attribute data is changed by an
          * Administrator.
-         * @see {@link MatterCoreSpecificationV1_0} § 9.10.7.1
          */
-        accessControlEntryChanged: Event(0, EventPriority.Info, {
-            ... AccessChangeEvent,
-            /** The latest value of the changed entry. */
-            latestValue: Field(3, AccessControlEntryT), /* nullable: true */
-        }), /* readAcl: AccessLevel.Administer */
+        accessControlEntryChanged: Event(0, EventPriority.Info, AccessChangeEvent(TlvAccessControlEntry)), /* readAcl: AccessLevel.Administer */
```
Comment (mfucci) [inline src/matter/cluster/AccessControlCluster.ts:169]: The original code was incorrect here...

**Comment by mfucci** [comment]
@Apollon77 Did you find time to review this?

**Comment by Apollon77** [comment]
Oohh overseen. Will do today

File: src/matter/cluster/GroupsCluster.ts [hunk 4]
```diff
@@ -111,37 +102,35 @@ export const GroupsCluster = Cluster({
...
-        nameSupport: Attribute(0, nameSupportBitmapT, { default: { groupNames: true } }),
+        nameSupport: Attribute(0, TlvNameSupportBitmap, { default: { groupNames: true } }),
     },
+
     /** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.3.7 */
     commands: {
         /**
          * The AddGroup command allows a client to add group membership in a particular group for the server endpoint.
-         *
-         * @see {@link MatterApplicationClusterSpecificationV1_0} § 1.3.7.1
```
Comment (Apollon77) [inline src/matter/cluster/GroupsCluster.ts:121]: why you removed the specs links here

Comment (Apollon77) [reply src/matter/cluster/GroupsCluster.ts:121]: or is it enough to have th general chapter on top?

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Only  sa small thing, rest wow ... a biggie but an awesome step forward

**Comment by mfucci** [comment]
All comments addressed!

**Review by Apollon77 — APPROVED** [review]
Approved. Thank you. I think I will have time next week to continue cluster efforts ... so perfect timing :-)

**Pull request merged**